launch_xml/launch_xml/entity.py

```diff
@@ -92,7 +92,7 @@ def get_attr(
         """
         attr_error = AttributeError(
             "Attribute '{}' of type '{}' not found in Entity '{}'".format(
-                name, data_type, self.type_name
+                name, data_type.__name__, self.type_name
```

Comment:
it's not reflected in the type annotation above, but `data_type` can be `None` apparently here and below :/ see the documentation for this method in the base class: https://github.com/ros2/launch/blob/6854fae626f2948139e1734e68d712cc1e0def1b/launch/launch/frontend/entity.py#L89

1. https://ci.ros2.org/job/ci_linux/24014/testReport/junit/test_launch_ros.test.test_launch_ros.frontend/test_component_container/test_launch_component_container_xml/
2. https://ci.ros2.org/job/ci_linux/24014/testReport/junit/test_launch_ros.test.test_launch_ros.frontend/test_node_frontend/test_launch_frontend_xml/

I'll open an issue to fix the type annotation.